src/components/CsvUpload/CsvUpload.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useRef, useState } from 'react'
+import { FileRejection, useDropzone } from 'react-dropzone'
+import { HStack, Spacer, VStack } from '../ui/Stack/Stack'
+import { P } from '../ui/Typography/Text'
+import UploadCloud from '../../icons/UploadCloud'
+import classNames from 'classnames'
+import { IconButton } from '../Button'
+import { DataState, DataStateStatus } from '../DataState/DataState'
+import { FileSpreadsheet } from 'lucide-react'
+import CloseIcon from '../../icons/CloseIcon'
+
+const validateCsvFile = (file: File) => {
+  const validExtensions = ['.csv']
+  const isValidExtension = validExtensions.some(ext =>
+    file.name.toLowerCase().endsWith(ext),
+  )
+
+  if (!isValidExtension) {
+    return 'File extension must end in .csv'
+  }
+
+  // Check MIME type if it exists
+  if (file.type && file.type !== 'text/csv' && file.type !== 'application/vnd.ms-excel') {
```

Comment:
recommend we support `text/plain` as well - if folks do edits in place i think there's surface area for it to get saved as pure text